### PR TITLE
Feature/clipboard sync

### PR DESCRIPTION
**Pull request by gtluszcz**
- Replacing text is now made with synchronous clipboard replacements so that we can be sure that  they are executed in order 
- Any error with clipboard will be indicated with notification (if possible)

### CODE REVIEW

File: src/main/modules/SnippetsManager.js [hunk 1]
```diff
@@ -2,18 +2,18 @@ const chars = require('./chars')
 const NativeKeymap = require('native-keymap')
 const _ = require('lodash')
 const Notification = require('./Notification')
+const clipboardy = require('clipboardy')
```
Comment (DCzajkowski) [inline src/main/modules/SnippetsManager.js]: This dependency should be injected via the constructor to preserve testability.

File: src/main/modules/SnippetsManager.js [hunk 2]
```diff
@@ -188,12 +188,31 @@ class SnippetsManager {
     }
 
     replace(value) {
-        const clipboardContent = this.clipboard.readText()
+        try {
+            // 1. store old clipboard data
```
Comment (DCzajkowski) [inline src/main/modules/SnippetsManager.js]: I don't think comments are necessary nor are adding any value. Let's write code that doesn't require comments ;)

File: src/main/modules/SnippetsManager.js [hunk 3]
```diff
@@ -188,12 +188,31 @@ class SnippetsManager {
...
+            const clipboardContent = clipboardy.readSync()
+            // 2. place value in clipboard
+            clipboardy.writeSync(value)
+            // 3. perform paste (takes 1-3 ms)
+            this.keyboardSimulator.keyTap('v', 'command')
+            // 4. place stored data back in clipboard
+            setTimeout(() => clipboardy.writeSync(clipboardContent), 50)
+            return true
```
Comment (DCzajkowski) [inline src/main/modules/SnippetsManager.js]: Returning a boolean may be a nice thing to do, but we are not using this returned information nowhere, so I don't see any value.

**Review by DCzajkowski — CHANGES_REQUESTED** [review]

File: src/main/modules/SnippetsManager.js (hunk 2 above)
Comment (gtluszcz) [reply src/main/modules/SnippetsManager.js]: Sure, as you wish

**Pull request merged**